ui/src/components/App.js

```diff
@@ -31,36 +38,43 @@ function PrivateOutlet() {
   const loggedIn = useSelector((state) => state.login.loggedIn);
   const location = useLocation();
   requireAuth();
-  return loggedIn ? <Outlet /> : <Navigate to="/login" state={{ from: location }} replace />;
+  return loggedIn ? (
+    <Outlet />
+  ) : (
+    <Navigate to="/login" state={{ from: location }} replace />
+  );
 }
 
-class App extends React.Component {
-  render() {
+function App(props) {
+  const { loggedIn } = props;
+
+  useEffect(() => {
     requireAuth();
+  }, []);
```

Comment:
`requireAuth` now runs asynchronously as a side effect, which is more idiomatic. No difference in practice.